Compilation is broken on main

### Describe the bug

The main branch fails to compile as of 99a4d06dfeeefa5dfa4eef58b955a73d51ebdbdf

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1-jboss-2:compile (default-compile) on project quarkus-core-deployment: Compilation failure: Compilation failure: 
[ERROR] /path.to/quarkus/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java:[610,25] cannot find symbol
[ERROR]   symbol:   variable isContainerBuild
[ERROR]   location: class io.quarkus.deployment.pkg.steps.NativeImageBuildStep.NativeImageInvokerInfo.Builder
[ERROR] /path.to/workspace/quarkus/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java:[610,24] illegal parenthesized expression
```

### Expected behavior

It compiles.

### Actual behavior

It doesnt compile.

### How to Reproduce?

mvn clean install -DskipTests

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java
@@ -110,6 +110,7 @@ public class NativeImageBuildStep {
                 .setRunnerJarName(runnerJar.getFileName().toString())
                 // the path to native-image is not known now, it is only known at the time the native-sources will be consumed
                 .setNativeImageName(nativeImageName)
+                .setContainerBuild(nativeConfig.isContainerBuild())
                 .build();
         List<String> command = nativeImageArgs.getArgs();
         try (FileOutputStream commandFOS = new FileOutputStream(outputDir.resolve("native-image.args").toFile())) {
@@ -201,6 +202,7 @@ public class NativeImageBuildStep {
                     .setRunnerJarName(runnerJarName)
                     .setNativeImageName(nativeImageName)
                     .setNoPIE(noPIE)
+                    .setContainerBuild(isContainerBuild)
                     .setGraalVMVersion(graalVMVersion)
                     .build();
 
@@ -474,6 +476,7 @@ public class NativeImageBuildStep {
             private Path outputDir;
             private String runnerJarName;
             private String noPIE = "";
+            private boolean isContainerBuild = false;
             private GraalVM.Version graalVMVersion = GraalVM.Version.UNVERSIONED;
             private String nativeImageName;
             private boolean classpathIsBroken;
@@ -521,6 +524,11 @@ public class NativeImageBuildStep {
 
             public Builder setNoPIE(String noPIE) {
                 this.noPIE = noPIE;
+                return this;
+            }
+
+            public Builder setContainerBuild(boolean containerBuild) {
+                isContainerBuild = containerBuild;
                 return this;
             }
 

